### PR TITLE
chore: incorporates index-based Merkle range proof in the NMT specification

### LINKED ISSUE #128: Explain normal Merkle inclusion and range proof in the  NMT specification
# Problem
The NMT wrapper offers standard Merkle tree inclusion and range proof functionalities that are inherited from the encapsulated NMT. However, the current wrapper specification refers to these features in the NMT specification, which does not explicitly explain them as standalone functionalities, but rather as part of namespace proofs. 

# Acceptance Criteria
To address this issue, the NMT specification should include two new sections to cover normal MT inclusion and range proof, and update the wrapper to reference to these sections accordingly.

### PR DESCRIPTION
**Pull request by staheri14**
## Overview
Closes #128 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by evan-forbes — DISMISSED** [review]
LGTM! :+1:

**Review by rootulp — DISMISSED** [review]

File: docs/spec/nmt.md [hunk 1]
```diff
@@ -143,6 +143,36 @@ Note that hash of the leaf does not have to be verified against the underlying m
 
 If the queried `nID` falls outside the namespace range of the tree root, or the namespace tree is empty, then an empty NMT proof is valid by definition.
 
+### Index-based Merkle Range Proof
```
Comment (adlerjohn) [inline docs/spec/nmt.md:146]: fmt: could you run https://github.com/DavidAnson/markdownlint with fixes on this?

Comment (staheri14) [reply docs/spec/nmt.md:146]: Sure, the issues are fixed now https://github.com/celestiaorg/nmt/pull/162/commits/218d517480d869eec94f28ea90633925a9e22439

**Review by rootulp — APPROVED** [review]

**Comment by staheri14** [comment]
In case there are no further comments or feedback provided by tomorrow, I plan to proceed with merging it.

**Review by evan-forbes — APPROVED** [review]
LGTM :+1:

**Pull request merged**